### PR TITLE
[Tests-Only] Update phpstan to latest version

### PR DESCRIPTION
**Pull request by dpakach**
Related - owncloud/QA#642

Update phpstan to 0.12

### CODE REVIEW

File: lib/Controller/PaperHiveController.php [hunk 1]
```diff
@@ -142,7 +142,7 @@ public function getPaperHiveBookDiscussionCount($dir, $filename) {
 			// Silently ignore discussions as this might indicate temporary unavailability
 			$disscussionCount = \count($paperHiveDiscussions['discussions']);
 		}
-		return new DataResponse($disscussionCount, Http::STATUS_OK);
+		return new DataResponse([$disscussionCount], Http::STATUS_OK);
```
Comment (dpakach) [inline lib/Controller/PaperHiveController.php:145]: This is causing the unit tests to fail.... But `DataResponse` should take array/object in its constructor, not int

@phil-davis  should we fix the unit test or ?

Comment (phil-davis) [reply lib/Controller/PaperHiveController.php:145]: Everywhere else in apps `DataResponse` is being passed an array. So this change is correct.
Fix the unit tests.

**Comment by phil-davis** [comment]
@mrow4a I see that you did code here in the past, e.g. #14 
`phpstan` complains that `DataResponse` should be passed an array as its first parameter of the constructor. But there is code here that passes a string or a count directly.

What to do?

I guess that something is consuming the API here. We can keep `phpstan` happy by passing the item as an array `[0 => 'stuff']`. But that is probably going to make the consumer of the API unhappy.

If we really do need to pass a string or count directly, then we might have to update the PHPdoc in core of `DataResponse` ?

**Comment by mrow4a** [comment]
It should be an array, and unit test needs to be adjusted.

**Comment by phil-davis** [comment]
> It should be an array, and unit test needs to be adjusted.

Is an array like in the current PR OK? (ends up with key 0, and value is the discussion count)
Or should the array have some key like 'data' or?

**Comment by mrow4a** [comment]
@phil-davis let me test this PR later just to be sure

File: lib/Controller/PaperHiveController.php (hunk 1 above)
Comment (phil-davis) [reply lib/Controller/PaperHiveController.php:145]: OK, looked again. This fix is good.

**Comment by phil-davis** [comment]
Also needs unit tests adjustment for phpunit8 - I will look.

**Pull request merged**